### PR TITLE
Addon-docs: Polish styling for 7.0

### PR DESCRIPTION
**Pull request by MichaelArestad**
## What I did

Adds a bit of polish and fixes the ArgsTable rendering in Safari. Huge props to @estelle for helping me out with this.

* Fixed ArgsTable rendering in Safari
* Updated vertical rhythm
* Updated headings
* Updated link icon

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/ui/blocks/src/blocks/mdx.tsx [hunk 1]
```diff
@@ -177,17 +179,14 @@ const HeaderWithOcticonAnchor: FC<HeaderWithOcticonAnchorProps> = ({
...
       >
-        <svg
-          viewBox="0 0 16 16"
-          version="1.1"
-          width="16"
-          height="16"
-          aria-hidden="true"
-          fill="currentColor"
-        >
+        <svg viewBox="0 0 14 14" width="14px" height="14px">
```
Comment (domyen) [inline code/ui/blocks/src/blocks/mdx.tsx:182]: why not import from our icon set instead?

File: code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx [hunk 2]
```diff
@@ -104,91 +103,58 @@ export const TableWrapper = styled.table<{
...
+        // I wasn't sure what kinds of content go in these tables so I was extra specific with selectors
+        // to avoid unexpected surprises.
+        ...(inAddonPanel
+          ? null
+          : {
+              filter:
                 theme.base === 'light'
-                  ? transparentize(0.035, theme.appBorderColor)
-                  : opacify(0.05, theme.appBorderColor),
+                  ? `drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.10))`
```
Comment (domyen) [inline code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx:120]: Do we still want to use `appBorderColor`?

File: code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx [hunk 3]
```diff
@@ -104,91 +103,58 @@ export const TableWrapper = styled.table<{
...
+        },
+        '> tr:last-of-type > td:first-of-type': {
+          borderBottomLeftRadius: theme.appBorderRadius,
+        },
+        '> tr:last-of-type > td:last-of-type': {
+          borderBottomRightRadius: theme.appBorderRadius,
         },
       },
-      // End finicky table styling
+      // End awesome table styling
```
Comment (domyen) [inline code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx:157]: lol

**Review by domyen — CHANGES_REQUESTED** [review]

File: code/ui/blocks/src/blocks/mdx.tsx (hunk 1 above)
Comment (MichaelArestad) [reply code/ui/blocks/src/blocks/mdx.tsx:182]: I'll give it a go.

File: code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx (hunk 2 above)
Comment (MichaelArestad) [reply code/ui/blocks/src/components/ArgsTable/ArgsTable.tsx:120]: No. It isn't used on the canvas drop shadows so I modified this to match the canvas shadow styles.

**Review by JReinhold — APPROVED** [review]
LGTM! 💪✨

**Pull request merged**